runway/exceptions.py

```diff
@@ -99,9 +105,14 @@ def __init__(self, response: dict[str, Any]) -> None:
                 that may not streamed.
 
         """
-        self.exit_code = response.get("StatusCode", 1)  # we can assume this will be > 0
-        error: dict[Any, Any] = response.get("Error") or {}  # value from dict could be NoneType
-        self.message = error.get("Message", "error message undefined")
+        self.exit_code = (
+            response.get("StatusCode", 1) if response else 1
+        )  # we can assume this will be > 0
+        error: dict[Any, Any] | None = (
+            response.get("Error") if response else {}
+        )  # value from dict could be NoneType
+        if error:
+            self.message = error.get("Message", "error message undefined")
```

Comment:
This can be done without changing the signature by storing the `response` on the object.

```suggestion
        self._response = response
        self.exit_code = response.get("StatusCode", 1)  # we can assume this will be > 0
        error: dict[Any, Any] = response.get("Error") or {}  # value from dict could be NoneType
        self.message = error.get("Message", "error message undefined")
```

Then, the `__reduce__` method would return self.__class__, (self._response,)